torch/lib/c10d/frontend.cpp

```diff
@@ -70,6 +70,61 @@ void Backend::registerBackend() {
   TORCH_CHECK(false, "Registering third-party backend is currently not supported by TorchScript-friendly c10d");
 }
 
+c10::intrusive_ptr<DistributedC10d> DistributedC10d::get() {
+  static c10::intrusive_ptr<DistributedC10d> singleton =
+      c10::make_intrusive<DistributedC10d>();
+
+  return singleton;
+}
+
+c10::intrusive_ptr<ProcessGroup> DistributedC10d::getProcessGroupByName(const std::string& name) const {
+  auto it = std::find_if(
+      pg_names_.begin(),
+      pg_names_.end(),
+      [&](const std::pair<c10::intrusive_ptr<ProcessGroup>, std::string>&
+              pg_name) { return pg_name.second == name; });
+
+  if (it == pg_names_.end()) {
+    std::stringstream error;
+    error << "Unable to find process group with name: ";
+    error << name;
+    error << " , instead we have ";
+    error << pg_names_.size() << " process groups: {";
+    for (const auto& pg : pg_names_) {
```

Comment:
it seems like if we are switch envs, and there's no pg created, user need to manually create pg that matches the name that they registered, shall we provide instructions to tell the user to do so?

Reply:
Yep we should, in fact we should write this logic for our current user.